llvm/lib/Support/SimpleTable.cpp

```diff
@@ -109,6 +109,16 @@ Error SimpleTable::replaceColumn(StringRef Name, const SimpleTable &Src,
   return Error::success();
 }
 
+Error SimpleTable::updateCellValue(int Row, StringRef ColName,
+                                   StringRef NewValue) {
+  if (getNumColumns() == 0)
+    return makeError("empty table");
+  if (Row > getNumRows())
+    return makeError("row index out of bounds");
+  Rows[Row][getColumnId(ColName)] = NewValue.str();
```

Comment:
Strictly speaking, we should probably check that `Row >= 0` as well, because it is defined as signed integer

Reply:
Absolutely! This has been added to the bounds check.

Reply:
Just in case it's relevant: there is a dedicated unit test for `SimpleTable` class validation - llvm/unittests/Support/SimpleTableTest.cpp.

Reply:
> 
> 
> Just in case it's relevant: there is a dedicated unit test for `SimpleTable` class validation - llvm/unittests/Support/SimpleTableTest.cpp.

I have added a test for `SimpleTable::updateCellValue` in that unittest. Are you suggesting that I add some negative testing as well?

Reply:
I'll leave it to you and @AlexeySachkov to decide.

I just noticed that I was added to this review as a code owner for this file, although I think it should be owned by @AlexeySachkov. I've opened https://github.com/intel/llvm/pull/3766 to fix the code owner of this file.